components/Dialogs/src/MDCAlertController.m

```diff
@@ -177,25 +162,14 @@ - (void)setMessage:(NSString *)message {
 
 - (void)addAction:(MDCAlertAction *)action {
   [_actions addObject:[action copy]];
+  if (self.alertView) {
+    [self.alertView addActionButtonTitle:action.title selector:@selector(actionButtonPressed:)];
 
-  MDCFlatButton *actionButton = [[MDCFlatButton alloc] initWithFrame:CGRectZero];
-  actionButton.mdc_adjustsFontForContentSizeCategory = self.mdc_adjustsFontForContentSizeCategory;
-  [actionButton setTitle:action.title forState:UIControlStateNormal];
-  // TODO(#1726): Determine default text color values for Normal and Disabled
-  CGRect buttonRect = actionButton.bounds;
-  buttonRect.size.height = MAX(buttonRect.size.height, MDCDialogActionButtonHeight);
-  buttonRect.size.width = MAX(buttonRect.size.width, MDCDialogActionButtonMinimumWidth);
-  actionButton.frame = buttonRect;
-  [actionButton addTarget:self
-                   action:@selector(actionButtonPressed:)
-         forControlEvents:UIControlEventTouchUpInside];
-  [self.actionsScrollView addSubview:actionButton];
-
-  [self.actionButtons addObject:actionButton];
+    self.preferredContentSize =
+        [self.alertView calculatePreferredContentSizeForBounds:CGRectInfinite.size];
 
-  self.preferredContentSize = [self calculatePreferredContentSizeForBounds:CGRectInfinite.size];
-
-  [self.view setNeedsLayout];
+    [self.alertView setNeedsLayout];
+  }
 }
 
 - (BOOL)mdc_adjustsFontForContentSizeCategory {
```

Comment:
`return self.alertView.mdc_adjustsFontForContentSizeCategory` we can remove the additional backing store BOOL in the top of the file.

Reply:
Unfortunately that won't work if alertView hasn't been instantiated yet.